### PR TITLE
move Kotlin/Native metadata deserialization + KLIB reader to a separate module

### PR DESCRIPTION
**Pull request by ddolovov**
move Kotlin/Native metadata deserialization + KLIB reader to a separate module

### CODE REVIEW

File: extracted/konan.serializer/src/org/jetbrains/kotlin/serialization/konan/interop/InteropUtils.kt [hunk 1]
```diff
@@ -0,0 +1,59 @@
+package org.jetbrains.kotlin.serialization.konan.interop
+
+import org.jetbrains.kotlin.descriptors.ModuleDescriptor
+import org.jetbrains.kotlin.descriptors.PackageFragmentDescriptor
+import org.jetbrains.kotlin.descriptors.konan.interop.InteropFqNames
+import org.jetbrains.kotlin.konan.library.KonanLibraryReader
+import org.jetbrains.kotlin.name.FqName
+import org.jetbrains.kotlin.serialization.konan.KonanPackageFragment
+
+fun createInteropLibrary(reader: KonanLibraryReader): InteropLibrary? {
```
Comment (olonho) [inline extracted/konan.serializer/src/org/jetbrains/kotlin/serialization/konan/interop/InteropUtils.kt]: Why this one goes to Kotlin? It has nothing to do with it.

Comment (ddolovov) [reply extracted/konan.serializer/src/org/jetbrains/kotlin/serialization/konan/interop/InteropUtils.kt]: It's needed for reading interop KLIBs.

For example, `InteropLibraryImpl` provides `ExportedForwardDeclarationsPackageFragmentDescriptor` for every package fragment in KLIB.

Comment (olonho) [reply extracted/konan.serializer/src/org/jetbrains/kotlin/serialization/konan/interop/InteropUtils.kt]: This shall be somehow rethought. This code does not belong to Kotlin at all.

File: extracted/konan.serializer/src/org/jetbrains/kotlin/serialization/konan/interop/InteropDescriptors.kt [hunk 2]
```diff
@@ -1,22 +1,5 @@
...
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package org.jetbrains.kotlin.backend.konan.descriptors
+package org.jetbrains.kotlin.serialization.konan.interop
```
Comment (SvyatoslavScherbina) [inline extracted/konan.serializer/src/org/jetbrains/kotlin/serialization/konan/interop/InteropDescriptors.kt]: The removed license header seems to be important.

**Pull request closed without merging**